src/main/java/cow/parser/Parser.java

```diff
@@ -0,0 +1,170 @@
+
+/**
+ * Class to handle the parsing of input commands.
+ **/
+public class Parser {
+    public static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
+
+    /**
+     * Parses the string input and returns a command that can be executed
+     * to carry out its task.
+     *
+     * @param userInput full user input from the user.
+     * @return command to be carried out.
+     * @throws CowExceptions used to print any exception that might happen.
+     */
+    public static Command parse(String userInput) throws CowExceptions {
+        Matcher matcher = BASIC_COMMAND_FORMAT.matcher(userInput.trim());
+
+        if (!matcher.matches()) {
+            return new IncorrectCommand(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                    HelpCommand.MESSAGE_USAGE));
+        }
+
+        final String commandWord = matcher.group("commandWord");
+        final String arguments = matcher.group("arguments");
+
+        switch (commandWord) {
+        case ListCommand.COMMAND_WORD:
+            return new ListCommand();
+        case ByeCommand.COMMAND_WORD:
+            return new ByeCommand();
+        case TodoCommand.COMMAND_WORD:
+            return prepareTodo(arguments);
+        case DeadlineCommand.COMMAND_WORD:
+            return prepareDeadline(arguments);
+        case EventCommand.COMMAND_WORD:
+            return prepareEvent(arguments);
+        case MarkCommand.COMMAND_WORD:
+            return new MarkCommand(getMarkUnmarkInt(arguments));
+        case UnmarkCommand.COMMAND_WORD:
+            return new UnmarkCommand(getMarkUnmarkInt(arguments));
+        case DeleteCommand.COMMAND_WORD:
+            return new DeleteCommand(getMarkUnmarkInt(arguments));
+        case DueCommand.COMMAND_WORD:
+            return prepDue(arguments);
+        case FindCommand.COMMAND_WORD:
+            if (arguments == null || arguments.isEmpty()) {
+                return new IncorrectCommand(String.format(Messages.MESSAGE_INVALID_COMMAND_FORMAT,
+                        FindCommand.MESSAGE_USAGE));
+            }
+            return new FindCommand(arguments);
+        default:
+            return new HelpCommand();
+        }
+    }
+
+    /**
+     * Parses the index used for marking, unmarking and delete commands.
+     *
+     * @param args provided by command.
+     * @return the parse int.
+     * @throws CowExceptions exceptions due to invalid index.
+     */
+    private static int getMarkUnmarkInt(String args) throws CowExceptions {
+        try {
+            return Integer.parseInt(args.trim());
+        } catch (NumberFormatException e) {
+            throw new CowExceptions(Messages.INDEX_INVALID);
+        }
+    }
+
+    /**
+     * Preps the argument to create the due command.
+     *
+     * @param args provided beside command.
+     * @return a due command.
+     */
+    private static Command prepDue(String args) {
```

Comment:
I think it's better to put the full word "prepare" in the method name

```suggestion
    private static Command prepareDue(String args) {
```